### ISSUE TITLE
Code highlighting?

### ISSUE THREAD

**Issue by spapas** [glowing-bear/glowing-bear#1052]
Could we add a switch to enable code highlighting using backticks (``) ? For example somebody could write 

> Here's my code snippet: \`this.is.a.code = snippet\`

and the \`this.is.a.code = snippet\` would be highlighted. From what I see in other clients (like irccloud) it just puts it in a &lt;pre&gt; html tag (with a monospaced font and a border). 

I know that there are various javascript libraries for code highlighting (f.e https://highlightjs.org/) but i'm not sure these will be actually useful in this case since they need to know the language in order to format it properly and inline snippets are small and most users wouldn't want to actually define the language for their snippet. 

This can also be enabled using a switch in settings like the latex math rendering .

**Comment by lorenzhs** [comment]
Since highlightjs' homepage states "automatic language detection" as a feature, I guess this could work. This would definitely be cool to have!

**Comment by spapas** [comment]
Hello @lorenzhs I tried using highlight.js but I don't like the results that much :| The language detection isn't working that good and it is actually recommended to specify it for small snippets: https://github.com/highlightjs/highlight.js/issues/1830 

Well this is something expected; how could anybody know that print("Hello"); is Javascript or C ?

In any case, I also had various problems trying to style the highlighted snippets for each line. The default styles for highlightjs are to be used with blocks and not inline (as we'd need in this case).

As a solution I recommend my initial proposal of just adding a code/pre around the backticked snippets. I tried it a bit and it worked fine.
